gipc/gipc.py

```diff
@@ -1058,6 +1073,27 @@ def put(self, o):
             self._write(struct.pack("!i", len(bindata)) + bindata)
 
 
+def reduce_GIPCReader(reader):
+    df = mpreduction.DupFd(reader._fd)
+    return (rebuild_GIPCReader, (df, reader._decoder))
+
+def rebuild_GIPCReader(df, decoder):
+    fd = df.detach()
+    return _GIPCReader(fd, decoder)
+
+mpreduction.register(_GIPCReader, reduce_GIPCReader)
+
+def reduce_GIPCWriter(writer):
+    df = mpreduction.DupFd(writer._fd)
+    return (rebuild_GIPCWriter, (df, writer._encoder))
+
+def rebuild_GIPCWriter(df, _encoder):
+    fd = df.detach()
+    return _GIPCWriter(fd, _encoder)
+
+mpreduction.register(_GIPCWriter, reduce_GIPCWriter)
```

Comment:
basically just copied this form from [connection.py](https://github.com/python/cpython/blob/main/Lib/multiprocessing/connection.py#L972-L978) Would probably need to add the windows way of doing this as well.

Reply:
> Would probably need to add the windows way of doing this as well.

Hope you also read through my old commentary in gipc for how I had chosen to do things on Windows ("stealing handles").

I learned a lot back then!